src/redux/books/books.js

```diff
@@ -1,23 +1,78 @@
 const ADD_BOOK = 'bookStore/books/ADD_BOOK';
 const REMOVE_BOOK = 'bookStore/books/REMOVE_BOOK';
+const FETCH_BOOK = 'bookStore/books/FETCH_BOOK';
 
 const initialState = [];
 
-export const addBook = (payload) => ({
+const returnBook = (payload) => ({
   type: ADD_BOOK,
   payload,
 });
-export const removeBook = (payload) => ({
-  type: REMOVE_BOOK,
-  payload,
-});
+
+export const addBook = (newBook) => async (dispatch) => {
+  const response = await fetch(
+    'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/kQDJnBa8pVnQ9r9R8wZR/books/',
+    {
+      method: 'POST',
+      headers: { 'Content-type': 'application/json' },
+      body: JSON.stringify(newBook),
+    },
+  );
+
+  if (response.ok) {
+    dispatch(returnBook(newBook));
+  }
+};
+
+export const removeBook = (id) => async (dispatch) => {
+  const body = {
+    item_id: id,
+  };
+  const response = await fetch(`https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/kQDJnBa8pVnQ9r9R8wZR/books/${id}`,
+    {
+      method: 'DELETE',
+      headers: { 'content-type': 'application/json' },
+      body: JSON.stringify(body),
+    });
+  if (response.ok) {
+    dispatch({
+      type: REMOVE_BOOK,
+      id,
+    });
+  }
+};
+
+export const fetchBook = () => async (dispatch) => {
+  const response = await fetch(
+    'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/kQDJnBa8pVnQ9r9R8wZR/books',
+  );
+  const result = await response.json();
+  const loadData = Object.entries(result);
+  loadData.map(([key, value]) => {
+    const { title, category } = value[0];
+    return {
+      item_id: key,
+      title,
+      category,
+    };
+  });
```

Comment:
There are some issues with this **`fetchBook`** function that makes it unable to store books correctly. I will highlight the issues below and provide a suggestion below.
- [ ] Your book listing on the page does not include the title, author, and category I added (see first 3 images)
- [ ] You app does not really delete anything from the API. I can add books, but when I remove one book and refresh the deleted book returns to the page (see the 3 images).

  Book is added | Book is "deleted" | Page is refreshed, calling `fetchBook`
  :---:|:---:|:---:
  ![francis-1](https://user-images.githubusercontent.com/36542986/157446392-3e0820e7-dfb5-4060-b87b-7342d61936e7.jpg) | ![francis-2](https://user-images.githubusercontent.com/36542986/157446396-a0aea21c-c6d3-42a3-943f-0b4d8cc403bc.jpg) | ![francis-3](https://user-images.githubusercontent.com/36542986/157446397-ea9fc4af-ac30-4fdd-bba5-a236add52d0b.jpg)

  ### Last image
  ![francis-last](https://user-images.githubusercontent.com/36542986/157446400-a38a1279-19dd-4316-956b-fe56cd08661a.jpg)
<hr>

  ## SUGGESTION:
  I checked your **book** item, there is really no such key as **`item_id`** (see the last image). This is because of how your book is mapped. You need to explicitly return the mapped book into a variable (here, I named it `books`) and dispatch that variable. This way book details are displayed and books can be deleted from the API.
  ```
  const loadData = Object.entries(result);
  const books = loadData.map(([key, value]) => {
    const { title, category } = value[0];
    return {
      item_id: key,
      title,
      category,
    };
  });

  if (books) {
    dispatch({
      type: FETCH_BOOK,
      payload: books,
    });
  }
  ```